src/steps/smart-campaign-add-lead.ts

```diff
@@ -35,12 +58,21 @@ export class AddLeadToSmartCampaignStep extends BaseStep implements StepInterfac
         return this.error("Can't add %s to %s: found %d matching campaigns", [email, campaign, campaigns.length]);
       }
 
-      const lead: any = await this.client.findLeadByEmail(email);
-      const result = await this.client.addLeadToSmartCampaign(campaigns[0].id.toString(), lead.result[0]);
-      if (result.success) {
-        return this.pass('Successfully added lead %s to smart campaign %s', [email, campaign]);
+      const campaignRecord = this.keyValue('campaign', 'Smart Campaign', campaigns[0]);
+
+      const leadResponse: any = await this.client.findLeadByEmail(email);
+
+      if (!leadResponse.result[0]) {
+        return this.fail('Could not find lead %s', [email], [campaignRecord]);
```

Comment:
- [ ] Should be `error`, since it is a precondition that is unmet (a lead existing for the given email address)